src/components/product-page/ImageCarousel.vue

```diff
@@ -0,0 +1,72 @@
+<script setup lang="ts">
+  import { ref } from "vue"
+  import IconMdi from "@/components/IconMdi.vue"
+  import { mdiChevronLeft, mdiChevronRight } from "@mdi/js"
+
+  const props = defineProps<{ images: string[] }>()
+  const activeIndex = ref(0)
+  const translate = ref(0)
+  const changeImage = (number: number) => {
+    translate.value = number / Math.abs(number)
+    activeIndex.value =
+      (((activeIndex.value + number) % props.images.length) +
+        props.images.length) %
+      props.images.length
+  }
+
+  const imageLoaded = ref(false)
+</script>
+
+<template>
+  <div class="relative">
+    <TransitionGroup
+      name="carousel"
+      tag="div"
+      class="relative w-full overflow-hidden rounded-2xl drop-shadow-lg">
+      <img
+        v-for="(image, index) in images"
+        @load="imageLoaded = true"
+        :src="image"
+        class="w-full"
+        :class="[
+          activeIndex !== index ? 'absolute top-0' : '',
+          !imageLoaded ? ' aspect-video bg-darkBackground' : ''
+        ]"
+        v-show="activeIndex === index"
+        :data-index="index"
+        :key="index" />
+    </TransitionGroup>
+    <icon-mdi
+      :icon="mdiChevronLeft"
+      :size="48"
+      stroke="hsla(0deg,0%,50%,0.5)"
```

Comment:
I was too lazy to set the stroke on everything, so I made it so you can use text color to change the stroke color